### PR TITLE
Enable nullability in TableLayout

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in TableLayout.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9840)

### CODE REVIEW

**Comment by Epica3055** [comment]
looks good, just some typos and format

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.cs [hunk 1]
```diff
@@ -391,7 +388,7 @@ private static bool xAssignRowsAndColumns(ContainerInfo containerInfo, LayoutInf
                 }
 
                 //set the rowStart and columnStart to fixedElement's specifed position
```
Comment (Epica3055) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/TableLayout.cs:390]: some typos.
```suggestion
                //set the rowStart and columnStart to fixedElement's specified position
```
And other parts of this file.



TableLayoutSettings.cs


``` c#
            Debug.Assert(!colSpecified || GetColumn(element) == column, "column position shoule equal to what we set");
            Debug.Assert(!rowSpecified || GetRow(element) == row, "row position shoule equal to what we set");
```

`shoule`

**Comment by Epica3055** [comment]
TableLayoutSettings.cs  line 65

``` c#
    /// <summary> internal as this is a TableLayoutPanel feature only </summary>
    [DefaultValue(TableLayoutPanelCellBorderStyle.None)]
    [SRCategory(nameof(SR.CatAppearance))]
    [SRDescription(nameof(SR.TableLayoutPanelCellBorderStyleDescr))]
```
to
``` c#
    /// <summary>
    ///  internal as this is a TableLayoutPanel feature only
    /// </summary>
    [DefaultValue(TableLayoutPanelCellBorderStyle.None)]
    [SRCategory(nameof(SR.CatAppearance))]
    [SRDescription(nameof(SR.TableLayoutPanelCellBorderStyleDescr))]
```

**Comment by Epica3055** [comment]
TableLayoutSettings.TableLayoutSettingsStub.cs

line 55
```c#
if (WindowsFormsUtils.SafeCompareStrings(name, controlName as string, /* ignoreCase = */ false))
```
please apply same strategy as you did in this PR

**Comment by Epica3055** [comment]
looks good. but it seems you missed one typo
TableLayoutSettings.cs line478


``` c#
            Debug.Assert(!colSpecified || GetColumn(element) == column, "column position shoule equal to what we set");
            Debug.Assert(!rowSpecified || GetRow(element) == row, "row position shoule equal to what we set");
```

`shoule`

**Review by Epica3055 — APPROVED** [review]

**Comment by gpetrou** [comment]
@dreddy-work maybe we could merge this one?

**Comment by dreddy-work** [comment]
> @dreddy-work maybe we could merge this one?

I will be going over NRT PRs next week.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**